autoarray/plot/mat_wrap/mat_plot.py

```diff
@@ -319,7 +319,7 @@ def plot_yx(
         y: Union[np.ndarray, List, Array1D],
         visuals_1d: Visuals1D,
         auto_labels: AutoLabels,
-        x: Optional[Union[np.ndarray, List, Array1D]] = None,
+        x: Optional[Union[np.ndarray, Iterable, List, Array1D]] = None,

```

Comment:
Would one of the types in this Union comprise all of the functionality expected of x?

Reply:
Possibly, but one still gets lint errors if you use for example an Iterable when you only type using np,ndarray

Reply:
np.ndarray implements iterable so if you use the Iterable type that might be sufficient. I'm not actually looking at the code but it depends on how you use the variable; if you just iterate through x then it being of type Iterable would suffice.

There's a rule where you should make your argument types as general as possible and your return types as specific as possible. This makes composition of functions more flexible.